presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -345,29 +341,59 @@ private QueryResult executeQueryControl()
             }
         }
         finally {
-            int retry = 0;
-            QueryResult tearDownResult;
-            do {
-                tearDownResult = tearDown(query, controlPostQueryResults, controlPostquery -> executeQuery(controlGateway, controlUsername, controlPassword, queryPair.getControl(), controlPostquery, controlTimeout, sessionProperties));
-                if (tearDownResult.getState() == State.SUCCESS) {
-                    break;
-                }
-                try {
-                    TimeUnit.MINUTES.sleep(1);
-                }
-                catch (InterruptedException e) {
-                    Thread.currentThread().interrupt();
-                    break;
-                }
-                retry++;
+            if (!skipDroppingTemporaryTables) {
+                QueryResult tearDownResult = tearDownQueryControl();
+                // if teardown is not successful the query fails
+                queryResult = tearDownResult.getState() == State.SUCCESS ? queryResult : tearDownResult;
             }
-            while (retry < controlTeardownRetries);
-            // if teardown is not successful the query fails
-            queryResult = tearDownResult.getState() == State.SUCCESS ? queryResult : tearDownResult;
         }
         return queryResult;
     }
 
+    private QueryResult tearDownQueryTest()
+    {
+        int retry = 0;
+        QueryResult tearDownResult;
+        do {
+            tearDownResult = tearDown(queryPair.getTest(), testPostQueryResults, testPostquery -> executeQuery(testGateway, testUsername, testPassword, queryPair.getTest(), testPostquery, testTimeout, sessionProperties));
+            if (tearDownResult.getState() == State.SUCCESS) {
+                break;
+            }
+            try {
+                TimeUnit.MINUTES.sleep(1);
+            }
+            catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                break;
+            }
+            retry++;
+        }
+        while (retry < testTeardownRetries);
+        return tearDownResult;
+    }
+
+    private QueryResult tearDownQueryControl()
```

Comment:
These methods are basically same. Add a helper method that they both call. The method will be easier to read since the variable names will be shorter (no test/control in all the names) and it will be easier to verify that we used the correct test/control variables for each one.